_drafts/wsl2-firmware-development.md

```diff
@@ -0,0 +1,655 @@
+  Seriously breaking my install.  Adding symlinks for OneDrive.  It’s been working
+  fine for a few months now, but for all I know I’m out of spec in a weird way that
+  will cause problems.  So, to quote Linus Torvalds, “F@&# you, \[Microsoft\]!”
+- 📎Unsettling paperclip mascot!
+
+#### VMWare Workstation Pro
+
+⭐⭐⭐
+
+
+Pros:
+- With a fully featured VM, you get most of the pros of Linux with access to Window’s strong points when you need them.
+- USB forwarding, including BLE, works well because you can run a full Linux distro of your choice with all the kernel modules that you need.
+- SSH to a headless VM allows for a snappy, low-latency, experience in your
+  editor of choice.
+
+Cons:
+- Setup and maintenance.  VMs are complicated, and setting up and maintaining the VM takes time and research.  A common example is the allocation of disk space.  Generally, you will need to allocate the space for your guest OS up front.  It can be hard to predict exactly how much space you really need, so you may end up needing to expand the virtual disk at some point, requiring steps to be taken from both the host and guest.
+
+#### WSL2
+
+⭐⭐⭐⭐
+
+Pros:
+- You get most of the pros of a Linux environment and all of the pros of the Windows host.
+- Resource allocation is automatic.  You can fiddle with CPU and RAM allocation, but you shouldn’t need to.  Storage is automatically resized as needed, so you never need to worry about running out of space for your WSL2 environment.
+- Setup is very simple — you can follow Intercreate’s tutorial below — and supported by Microsoft.
+
+Cons:
+- Minimal choice of distros and fewer options for customizing your installation and kernel.  That’s not to say that you _can’t_, but rather that you will stray outside of the Microsoft support umbrella by doing so.
+- BLE, and probably other kernel modules that I haven’t thought of, are not enabled in the default WSL2 configuration.
+- Performance is not as good as running Linux on bare metal.
+
+### Conclusion
+
+Ultimately, choosing a development environment depends on what helps you to
+provide the most value to your clients.  Decisions about security, performance,
+features, and stability must be weighed distinctly within each organization or
+project.  Personally, I’ve found a nice balance by using WSL2 on Windows 11. 
+I get all of the security, stability, compatibility, and support of Windows while
+being able to use my preferred tools in a Linux environment.  My primary concern,
+performance, has been eased by quantifying that a WSL2 Zephyr build is only ~12% slower
+than bare metal Linux.  If you compile big Zephyr projects with 40s build times - 
+_pristine instead of incremental_ - 20 times a day (I don’t), WSL2 costs you an extra 1m 30s
+of compile time.  But what does maintaining desktop Linux cost each day?  How
+much time do you have for fixing ALSA, compiling a tool that didn’t have a binary
+release, fiddling with WINE, or setting up a Windows VM?
```

Comment:
most of these aren't much of a hindrance anymore. The Windows VM one... not beginner friendly with QEMU/KVM.
Obviously, YMMV.

Reply:
Yeah, this is where I worry about my opinion vs fact.  My opinion is that Windows is much more stable the desktop Linux.  My evidence for this is only anecdotal; I've found my sometimes use of Ubuntu Desktop unstable (hard crashes), unsupported, etc etc,.

1) Linux has no market share
2) it has a ton of different distros and configurations - so even less market share for each if you say ubuntu + gnome, or debian + kde etc - we're talking fractions of a percent.
3) why would Windows (or any distro) share metrics about stability?  Wouldn't they all just say that they're perfect?

I think the right approach might be to remove this comment and let people draw their own conclusions.